lib/dialects/mariadb/data-types.js

```diff
@@ -92,6 +93,17 @@ module.exports = BaseTypes => {
         this.sqlType = this.type;
       }
     }
+    static parse(value) {
+      value = value.buffer();
+      // Empty buffer, MySQL doesn't support POINT EMPTY
+      // check, https://dev.mysql.com/worklog/task/?id=2381
+      if (!value || value.length === 0) {
+        return null;
+      }
+      // For some reason, discard the first 4 bytes
```

Comment:
Why?

Reply:
Seems same logic for MySQL applies for MariaDB

https://mariadb.com/kb/en/st_isempty/

> Since MariaDB and MySQL do not support GIS EMPTY values such as POINT EMPTY

Reply:
I mean why discard the first 4 bytes?

Reply:
First 4 bytes in MySQL and MariaDB represents SRID metadata. This type is known as EWKTB. 
While WKB corresponds to POINT(1 2);  EWKB corresponds to SRID=4326;POINT(1 2) kind of information.
Strangely wkx library is having a hard time parsing MySQL field of Geometry with SRID metadata. SRID is also included in WKTB so after cutting out SRID as metadata , we can parse WKTB and get the object as requested.

This cutting out thing is done way before this bug fix in MySQL dialect. So this PR has nothing to do with it, just using that approach. Maybe another look is required to change that.

By the way #11616 is not relevant with this PR. So removing it

Reply:
Ah, understood, thank you very much :+1: